### PR TITLE
Exercise 9

### PR DESCRIPTION
**Pull request by kallya21**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: .github/workflows/ci-pipeline.yml [hunk 1]
```diff
@@ -18,11 +18,27 @@ jobs:
     - run: docker build --target test --tag my-test-image .
     - run: docker run --env-file .env.test  my-test-image
 
+  deploy:
+    name: Build and deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.event_name == 'push' && github.ref == 'refs/heads/exercise-9'
```
Comment (JackMead) [inline .github/workflows/ci-pipeline.yml:25]: Don't forget to update this to main once you're merging 😄

File: .github/workflows/ci-pipeline.yml [hunk 2]
```diff
@@ -18,11 +18,27 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Login to Docker Hub
+      uses: docker/login-action@v2
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
+        password: ${{ secrets.DOCKER_PASSWORD }}
+    - run: docker build --target production --tag kallya21/todo-app:latest .
+    - run: docker push kallya21/todo-app:latest
+    - run: curl -dH -X POST '${{ secrets.AZURE_WEBHOOK_URL }}'
```
Comment (JackMead) [inline .github/workflows/ci-pipeline.yml:35]: Nice work getting this in place in the end - for my records I'm assuming this is an unescaped URL?

Comment (JackMead) [reply .github/workflows/ci-pipeline.yml:35]: A nice bonus to add to the curl command here can be a `--fail` argument - without this curl reports success if the request gets a response, even if that response is a 4xx/5xx response!

**Review by JackMead — APPROVED** [review]
Nice work, I can see your pipeline is happily building & pushing your image, and the webhook is correctly notifying Azure now! 😄 Very happy for you to merge this & the M8 docs when you're ready

File: .github/workflows/ci-pipeline.yml (hunk 2 above)
Comment (kallya21) [reply .github/workflows/ci-pipeline.yml:35]: Yes, it's an unescaped URL

**Pull request merged**